tools/content/fetch_messages.dart

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env dart
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+// Avoid any Flutter-related dependencies so this can be run as a CLI program.
+import 'package:args/args.dart';
+import 'package:http/http.dart';
+import 'package:ini/ini.dart' as ini;
+import 'package:zulip/api/backoff.dart';
+
+import 'model.dart';
+
+/// Fetch all public message contents from a Zulip server in bulk.
+///
+/// It outputs JSON entries of the message IDs and the rendered HTML contents in
+/// JSON Lines (https://jsonlines.org) format. The output can be used later to
+/// perform checks for discovering unimplemented features.
+///
+/// Because message IDs are only unique within a single server, the script
+/// names corpuses from each server differently (if --corpus-dir is specified).
+///
+/// This script is meant to be run via `tools/content/check-features`.
+///
+/// For more help, run `tools/content/fetch_message.dart --help`.
+///
+/// See also:
+/// * tools/content/unimplemented_features_test.dart, which runs checks against
+///   the fetched corpuses.
+void main(List<String> args) async {
+  final argParser = ArgParser();
+  argParser.addOption(
+    'config-file',
+    help: 'A zuliprc file with identity information including email, API key\n'
+          'and the Zulip server URL to fetch the messages from (required).\n\n'
+          'To get the file, see\n'
+          'https://zulip.com/api/configuring-python-bindings#download-a-zuliprc-file.',
+    valueHelp: 'path/to/zuliprc',
+  );
+  argParser.addOption(
+    'corpus-dir',
+    help: 'The directory to look for/store the corpus file.  If not given,\n'
+          'the script will write output to stdout.  Otherwise, this will\n'
```

Comment:
Writing to stdout seems like a bit of a pitfall for the unwary — it's going to send a lot of spew to one's terminal, right?

I imagine it was helpful when writing and debugging the script, and you were editing the code to control how much output it would send.

Fine to keep the option of stdout, but let's make it not the default. It could be a `--stdout` flag, incompatible with `--corpus-dir`.

Reply:
The thing about writing to stdout and keeping the script quiet was mainly for usability with pipes. I think we can eliminate this option and just make `--corpus-dir` mandatory.